packages/worker/src/index.ts

```diff
@@ -0,0 +1,186 @@
+   * @param handlers An array of handlers to register against this interface.
+   */
+  add(abi: string | readonly (string | Fragment | JsonFragment)[] | Interface, handlers: Array<HandlerDescription>) {
+    const abiInterface = toInterface(abi);
+
+    for (const handler of handlers) {
+      const fn = abiInterface.getFunction(handler.type);
+
+      this.handlers[Interface.getSighash(fn)] = {
+        type: fn,
+        func: handler.func,
+      };
+    }
+  }
+
+  /**
+   * Convenience function to construct an `itty-router` application object for the gateway.
+   * Example usage:
+   * ```javascript
+   * const ccipread = require('@chainlink/ccip-read-cf-worker');
+   * const server = new ccipread.Server();
+   * // set up server object here
+   * const app = server.makeApp('/');
+   * module.exports = {
+   *  fetch: function (request, _env, _context) {
+   *    return app.handle(request)
+   *  }
+   * };
+   * ```
+   * The path prefix to `makeApp` will have sender and callData arguments appended.
+   * If your server is on example.com and configured as above, the URL template to use
+   * in a smart contract would be "https://example.com/{sender}/{callData}.json".
+   * @returns An `itty-router.Router` object configured to serve as a CCIP read gateway.
+   */
+  makeApp(prefix: string): Router {
+    const app = Router();
+    app.get(prefix, () => new Response('hey ho!', { status: 200 }));
+    /*
+     * uncomment app.options for cors
+     * also wrap responses with cors wrapper
+     * e.g. return wrapCorsHeader(new Response('Invalid request format', { status: 400 }));
+     */
+    // app.options(`${prefix}:sender/:callData.json`, handleCors({ methods: 'GET', maxAge: 86400 }));
+    app.get(`${prefix}:sender/:callData.json`, this.handleRequest.bind(this));
+    // app.options(prefix, handleCors({ methods: 'POST', maxAge: 86400 }));
+    app.post(prefix, this.handleRequest.bind(this));
+    return app;
+  }
+
+  async handleRequest(req: IRequest) {
+    let sender: string;
+    let callData: string;
+
+    if (req.method === 'GET') {
+      sender = req.params.sender;
+      callData = req.params.callData;
+    } else {
+      const body = await req.json();
+      sender = body.sender;
+      callData = body.data;
+    }
+
+    if (!isAddress(sender) || !isBytesLike(callData)) {
+      return new Response('Invalid request format', { status: 400 });
+    }
+
+    try {
+      const response = await this.call({ to: sender, data: callData });
+      return new Response(JSON.stringify(response.body), {
+        status: response.status,
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+    } catch (e) {
+      return new Response(`Internal server error: ${(e as any).toString()}`, { status: 500 });
+    }
+  }
+
+  async call(call: RPCCall): Promise<RPCResponse> {
+    const calldata = hexlify(call.data);
+    const selector = calldata.slice(0, 10).toLowerCase();
```

Comment:
It might be good to refactor this into a function named `getFunctionSelector(calldata: string)`.